docs/dev/src/design/keys.md

```diff
@@ -2,9 +2,28 @@
 
 Document the different Keys in RisingWave.
 
+## Order Key
+
+The order key is user-specified, in SQL. For example:
+
+```sql
+create materialized view m1 as select * from t1 order by i, id;
+```
+
+The order key is `i, id`, which is the order of the columns in the `order by` clause.
+
+It is used to ensure locality of records in storage, and locality of record updates in streaming.
+To ensure storage locality, when deriving the storage key, we will use the order key as a prefix.
+To ensure streaming locality, we will use the order key as a prefix of the stream key.
+
+## Group Key
+
+Similar to the order key, the group key is also user-specified in SQL.
+We also apply the same derivations of group key in stream key, to ensure storage and streaming locality.
```

Comment:
I think order key and group key are not really the same level of concept as the following keys.

They are user-facing concepts. And the following are how things work internally.

Reply:
Maybe I'll include a section for them inside stream_key and storage_pk?
Since they are used in the derivation step.

Reply:
Indeed, a section about their derivation (maybe with examples) would be nice to have, and the user facing part should also go there.